jormungandr: handle Kraken priority

kadhikari opened the pull request:
1. Add an attribut "priority" in the table instance
   2.Choose kraken instance among all instances valide with largest priority value.

Commit:
tyr: attribut priority added in the table instance

Commit:
jormungandr: choose kraken instance by priority value

Commit:
gitignore: folders added

Commit:
jormungandr: init instance priority to 0 in tests

antoine-de commented on .gitignore:
```diff
@@ -28,3 +28,5 @@ debian/tmp
 debian/*.substvars
 debian/*.debhelper
 source/jormungandr/jormungandr/_version.py
+source/jormungandr/venv/*
```
I personally think that it's better to put those in you own personal .gitignore file (since it does not apply for everybody, and in your case you'll want this for every venv directory

kadhikari replied:
ok

antoine-de commented on source/jormungandr/jormungandr/instance_manager.py:
```diff
@@ -64,6 +64,10 @@ def instances_comparator(instance1, instance2):
         return jormun_bdd_instance1.is_free - jormun_bdd_instance2.is_free
 
+    #Here we choose the instance with greater priority.
+    if jormun_bdd_instance1.priority > jormun_bdd_instance2.priority:
```
hum I think you want to test first if they are different, then return the subtraction of the value

antoine-de commented on source/navitiacommon/navitiacommon/default_values.py:
```diff
@@ -92,6 +92,9 @@
 # night bus filtering parameter
 night_bus_filter_base_factor = 60*60  # seconds
 
+#priority
```
I think it deserves a better comment :wink:

kadhikari replied:
ok

antoine-de commented on source/tyr/migrations/versions/cf2becf2dc1_add_priority.py:
```diff
@@ -0,0 +1,26 @@
+
+
+def upgrade():
+    ### Attribut priority added in the table instance! ###
```
you can remove all the generated comments

kadhikari replied:
ok

antoine-de commented on source/jormungandr/jormungandr/interfaces/v1/test/journey_tests.py:
```diff
@@ -51,10 +52,10 @@ def __init__(self):
+            'equador': MockInstance(True, 'equador', 0),
+            'france': MockInstance(False, 'france', 0),
+            'peru': MockInstance(False, 'peru', 0),
+            'bolivia': MockInstance(True, 'bolivia', 0)
```
In all the tests the value is the same, so it's not tested, can you add a test with different values ?

Commit:
jormungandr: priority of kraken instances

- comments modified in script de migration
- Some tests added
- instance sorted by priority if greater then 0
else first non free followed by non-free

kadhikari commented:
@antoine-de  i made some modifications on your remarks

antoine-de commented on source/navitiacommon/navitiacommon/default_values.py:
```diff
@@ -92,6 +92,9 @@
 # night bus filtering parameter
 night_bus_filter_base_factor = 60*60  # seconds
 
+#priority value to be used to choose un instance kraken among valid instances
```
Maybe: 

priority value to be used to choose a kraken instance among valid instances. The greater it is, the more the instance will be chosen

antoine-de commented on source/jormungandr/jormungandr/interfaces/v1/test/journey_tests.py:
```diff
@@ -154,3 +158,36 @@ def test_multi_coverage_overlap_chose_non_free2(self):
         assert set([regions[0].name, regions[1].name]) == set([self.regions['france'].name, self.regions['peru'].name])
         assert set([regions[2].name, regions[3].name]) == set([self.regions['equador'].name, self.regions['bolivia'].name])
+
+    def test_multi_coverage_overlap_chose_with_non_free_and_priority(self):
```
great tests! :+1:

antoine-de commented:
apart from my minor remark, OK for me

kinnou02 commented:
the priority should probably be displayed in the status of an instance on jormungandr too.

Commit:
jormungandr: display instance priority jormungandr/status

Comments modified.

kadhikari commented:
@antoine-de  
priority added in /status.parameters and comments modified
